birthday/birthday.py

```diff
@@ -0,0 +1,259 @@
+            print(self.BDAY_INVALID())
+            await channel.send(self.BDAY_INVALID())
+        else:
+            await self.remove_user_bday(message.guild.id, author.id)
+            await self.get_date_config(message.guild.id, birthday.toordinal()).get_attr(author.id).set(year)
+            bday_month_str = birthday.strftime("%B")
+            bday_day_str = birthday.strftime("%d").lstrip("0")
+            await channel.send(self.BDAY_SET(bday_month_str + " " + bday_day_str))
+
+    @bday.command(name="list")
+    async def bday_list(self, ctx: Context):
+        """Lists birthdays
+
+        If a user has their year set, it will display the age they'll get after their birthday this year"""
+        message = ctx.message
+        await self.clean_bdays()
+        bdays = await self.get_guild_date_configs(message.guild.id)
+        this_year = datetime.date.today().year
+        embed = discord.Embed(title=self.BDAY_LIST_TITLE(), color=discord.Colour.lighter_grey())
+        for k, g in itertools.groupby(sorted(datetime.datetime.fromordinal(int(o)) for o in bdays.keys()),
+                                      lambda i: i.month):
+
+            value = "\n".join(date.strftime("%d").lstrip("0") + ": "
+                              + ", ".join("<@!{}>".format(u_id)
+                                          + ("" if year is None else " ({})".format(this_year - int(year)))
+                                          for u_id, year in bdays.get(str(date.toordinal()), {}).items())
+                              for date in g if len(bdays.get(str(date.toordinal()))) > 0)
+            if not value.isspace():
+                embed.add_field(name=datetime.datetime(year=1, month=k, day=1).strftime("%B"), value=value)
+        await message.channel.send(embed=embed)
+
+    async def clean_bday(self, guild_id: int, guild_config: dict, user_id: int):
+        guild = self.bot.get_guild(guild_id)
+        if guild is not None:
+            role = discord.utils.get(guild.roles, id=guild_config.get("role"))
+
+            await self.maybe_update_guild(guild)
+            member = guild.get_member(user_id)
+            if member is not None and role is not None and role in member.roles:
+
+                await member.remove_roles(role)
+
+    async def handle_bday(self, user_id: int, year: str):
+        embed = discord.Embed(color=discord.Colour.gold())
+        if year is not None:
+            age = datetime.date.today().year - int(year)
+            embed.description = self.BDAY_WITH_YEAR(user_id, age)
+        else:
+            embed.description = self.BDAY_WITHOUT_YEAR(user_id)
+        all_guild_configs = await self.config.all_guilds()
+        for guild_id, guild_config in all_guild_configs.items():
+            guild = self.bot.get_guild(guild_id)
+            if guild is not None:
+                member = guild.get_member(user_id)
+                if member is not None:
+                    role_id = guild_config.get("role")
```

Comment:
I would also dm the user and tell them happy birthday, but remember you can't use server emojis in dm's, so stick to the default ones.